spec/q-spec.js

```diff
@@ -1096,6 +1096,15 @@ describe("propagation", function () {
 });
 
 describe("all", function () {
+    it("rejects if not passed an array", function() {
+        var promise = Q.defer(),
+            willBeRejected = Q.all(promise);
+
+        willBeRejected.then(function() {
+            expect(willBeRejected.isRejected()).toBe(true);
```

Comment:
This will never execute, because it is a fulfillment callback.

Also, to get the test runner to recognize async tests, you need to return the promise.

So I think this would be better:

```js
return willBeRejected.then(
  function () {
    expect(true).toBe(false);
  },
  function (err) {
    expect(err.constructor).toBe(TypeError);
  }
);
```